Detectors/MUON/MID/Workflow/src/tracks-reader-workflow.cxx

```diff
@@ -0,0 +1,131 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file   MID/Workflow/src/tracks-reader-workflow.cxx
+/// \brief  DPL workflow to send MID tracks read from a root file
+/// \author Philippe Pillot, Subatech
+
+#include <algorithm>
+#include <cstring>
+#include <functional>
+#include <iterator>
+#include <memory>
+#include <stdexcept>
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/ControlService.h"
+#include "Framework/Task.h"
+
+#include "DPLUtils/RootTreeReader.h"
+
+#include "DataFormatsMID/ROFRecord.h"
+#include "DataFormatsMID/Track.h"
+
+using namespace o2::framework;
+using namespace o2::mid;
+
+class TrackSamplerTask
+{
+ public:
+  /// prepare the reader
+  void init(InitContext& ic)
+  {
+    auto inputFileName = ic.options().get<std::string>("infile");
+    mMinNumberOfROFsPerTF = ic.options().get<int>("repack-rofs");
+
+    mReader = std::make_shared<RootTreeReader>("midreco", inputFileName.c_str(), -1,
```

Comment:
just curious : why must it be a shared pointer ?

Reply:
I followed examples I found here and there, but it's true that the way I am defining the algorithm spec, with a task instead of using lambda, I do not need to pass the reader to the lambda so I could probably use a unique_ptr instead.